.make/test.mk

```diff
@@ -90,79 +116,154 @@ test-integration: prebuild-check clean-test-artifacts-integration $(COVERAGE_INT
+
+# Experimental:
 
 .PHONY: gocov-unit-annotate
 ## (EXPERIMENTAL) Show actual code and how it is covered with unit tests.
 ##                This target only runs the tests if the coverage file does exist.
-gocov-unit-annotate: prebuild-check $(GOCOV_BIN) $(COVERAGE_UNIT_PATH)
-	$(GOCOV_BIN) convert $(COVERAGE_UNIT_PATH) | $(GOCOV_BIN) annotate -
+gocov-unit-annotate: prebuild-check $(GOCOV_BIN) $(COV_PATH_UNIT)
+	@$(GOCOV_BIN) convert $(COV_PATH_UNIT) | $(GOCOV_BIN) annotate -
 
 .PHONY: .gocov-unit-report
-.gocov-unit-report: prebuild-check $(GOCOV_BIN) $(COVERAGE_UNIT_PATH)
-	$(GOCOV_BIN) convert $(COVERAGE_UNIT_PATH) | $(GOCOV_BIN) report
+.gocov-unit-report: prebuild-check $(GOCOV_BIN) $(COV_PATH_UNIT)
+	@$(GOCOV_BIN) convert $(COV_PATH_UNIT) | $(GOCOV_BIN) report
 
 .PHONY: gocov-integration-annotate
 ## (EXPERIMENTAL) Show actual code and how it is covered with integration tests.
 ##                This target only runs the tests if the coverage file does exist.
-gocov-integration-annotate: prebuild-check $(GOCOV_BIN) $(COVERAGE_INTEGRATION_PATH)
-	$(GOCOV_BIN) convert $(COVERAGE_INTEGRATION_PATH) | $(GOCOV_BIN) annotate -
+gocov-integration-annotate: prebuild-check $(GOCOV_BIN) $(COV_PATH_INTEGRATION)
+	@$(GOCOV_BIN) convert $(COV_PATH_INTEGRATION) | $(GOCOV_BIN) annotate -
 
 .PHONY: .gocov-integration-report
-.gocov-integration-report: prebuild-check $(GOCOV_BIN) $(COVERAGE_INTEGRATION_PATH)
-	$(GOCOV_BIN) convert $(COVERAGE_INTEGRATION_PATH) | $(GOCOV_BIN) report
+.gocov-integration-report: prebuild-check $(GOCOV_BIN) $(COV_PATH_INTEGRATION)
+	@$(GOCOV_BIN) convert $(COV_PATH_INTEGRATION) | $(GOCOV_BIN) report
 
 #-------------------------------------------------------------------------------
-# Test artifacts are two coverage files for unit and integration tests.
+# Test artifacts are coverage files for unit and integration tests.
 #-------------------------------------------------------------------------------
 
-$(COVERAGE_UNIT_PATH): prebuild-check
-	go test $(go list ./... | grep -v vendor) -v -coverprofile $(COVERAGE_UNIT_PATH) -covermode=$(COVERAGE_MODE)
+# The test-package function executes tests for a package and saves the collected
+# coverage output to a directory. After storing the coverage information it is
+# also appended to a file of choice (without the "mode"-line)
+#
+# Parameters:
+#  1. Test name (e.g. "unit" or "integration")
+#  2. package name "github.com/almighty/almighty-core/model"
+#  3. File in which to combine the output
+#  4. (optional) parameters for "go test" command
+define test-package
+$(eval TEST_NAME := $(1))
+$(eval PACKAGE_NAME := $(2))
+$(eval COMBINED_OUT_FILE := $(3))
+$(eval EXTRA_TEST_PARAMS := $(4))
+
+@mkdir -p $(COV_DIR)/$(PACKAGE_NAME);
+$(eval COV_OUT_FILE := $(COV_DIR)/$(PACKAGE_NAME)/coverage.$(TEST_NAME).mode-$(COVERAGE_MODE))
+@ALMIGHTY_DB_HOST=localhost go test $(PACKAGE_NAME) -v -coverprofile $(COV_OUT_FILE) -covermode=$(COVERAGE_MODE) -timeout 10m $(EXTRA_TEST_PARAMS);
```

Comment:
I'm in the middle of integrating https://github.com/spf13/viper for configuration file reading :)